src/DependencyInjection/Admin/AutoConfiguredAdminInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\DependencyInjection\Admin;
+
+interface AutoConfiguredAdminInterface
+{
+    /**
+     * Return an array wuth all parameters required to configure admin:
+     * [
+     *    code: ?string
+     *    class: string
+     *    controller: ?string
+     *    label: ?string
+     *    show_in_dashboard: bool
+     *    group: string
+     *    label_catalog: ?string
+     *    icon: ?string
+     *    on_top: bool
+     *    keep_open: bool
+     *    manager_type: string
+     * ]
+     *
+     * @return array
+     */
+    public static function getAdminConfiguration() : array;
```

Comment:
Rather than only one method returning an array, I think it could be better to have one method by key.
This would also be easier to override. WDYT ?

Reply:
I think this is gonna be too much verbose, @core23 propose php8 attribute and php annoation, i think this is the most convinent way to do this.
I'll upate PR in that way

Reply:
I think instead it could be less verbose and easier to override.

With array, you have to do
```
    use AutoConfiguredAdminTrait {
        AutoConfiguredAdminTrait::getAdminConfiguration as parentTraitAdminConfiguration;
    }
```
and then use array_merge with the right `keys`.

With 11 methods, I just need to override the one I need to. Which might be easier.

Also, if we prefer to use only one method like you did in your interface, we should return a valueObject rather than an array IMHO.

Reply:
Maybe a valueObject is nicer? Im not relly sure about the solution. Probably starting with the docs part gives us a better understanding of the usage, and avoid going back and forth with the implementation.